packages/flutter/lib/src/material/dialog.dart

```diff
@@ -467,6 +467,9 @@ class _DialogRoute<T> extends PopupRoute<T> {
 /// Returns a [Future] that resolves to the value (if any) that was passed to
 /// [Navigator.pop] when the dialog was closed.
 ///
+/// The dialog is pushed to the root navigator.
+/// Use `Navigator.of(context, rootNavigator: true)` to pop the dialog.
```

Comment:
How about:
```
/// The dialog route created by this method is pushed to the root navigator.
/// If the application has multiple [Navigator] objects, it may be necessary to
/// call `Navigator.of(context, rootNavigator: true).pop(result)` to close the
/// dialog rather just 'Navigator.pop(context, result)`.
```